app/javascript/components/automate-entry-points/index.jsx

```diff
@@ -0,0 +1,279 @@
+              id: newNode.id,
+              name: newNode.name,
+              children: [],
+              parent: element.id,
+              metadata: { domain_fqname: newNode.domain_fqname, fqname: newNode.fqname },
+            });
+          }
+        }
+      });
+      return new Promise((resolve) => {
+        if (element.children.length > 0) {
+          resolve();
+          return;
+        }
+        setTimeout(() => {
+          setData((value) =>
+            updateTreeData(value, element.id, newChildren));
+          resolve();
+        }, 1000);
+      });
+    });
+  };
+
+  const wrappedOnLoadData = async({ element }) => {
+    const nodeHasNoChildData = element.children.length === 0;
+    const nodeHasAlreadyBeenLoaded = nodesAlreadyLoaded.find(
+      (e) => e.id === element.id
+    );
+
+    await onLoadData({ element });
+
+    if (nodeHasNoChildData && !nodeHasAlreadyBeenLoaded) {
+      setNodesAlreadyLoaded([...nodesAlreadyLoaded, element]);
+
+      setTimeout(() => {
+      }, 1000);
+    }
+  };
+  wrappedOnLoadData.propTypes = {
+    element: PropTypes.objectOf({ children: PropTypes.array, id: PropTypes.number }).isRequired,
+  };
+  wrappedOnLoadData.defaultProps = {
+  };
+
+  const onSelect = (value) => {
+    if (value.isBranch === false && value.isSelected) {
+      data.forEach((node) => {
+        if (selectedNode && (node.id === selectedNode.element.id)) {
+          document.getElementById(node.id).style.backgroundColor = 'transparent';
+        }
+      });
+      document.getElementById(value.element.id).style.backgroundColor = 'rgba(0, 0, 0, 0.2)';
+      setSelectedNode(value);
+      setDisableSubmit(false);
+    }
+  };
+
+  const onExpand = ((value) => {
+    if (value.isExpanded && selectedNode && document.getElementById(selectedNode.element.id)) {
+      document.getElementById(selectedNode.element.id).style.backgroundColor = 'rgba(0, 0, 0, 0.2)';
+    }
+  });
+
+  const FolderIcon = ({ isOpen }) =>
+    (isOpen ? (
+      <FolderOpen16 className="icon" />
+    ) : (
+      <Folder16 className="icon" />
+    ));
+
+  FolderIcon.propTypes = {
+    isOpen: PropTypes.bool,
+  };
+  FolderIcon.defaultProps = {
+    isOpen: false,
+  };
+
+  const FileIcon = () => <Document16 className="icon" />;
+
+  return !isLoading && (
+    <Modal
+      open={showModal}
+      primaryButtonText={__('OK')}
+      secondaryButtonText={__('Cancel')}
+      onRequestSubmit={() => {
+        setSelectedValue(selectedNode);
+        setShowModal(false);
+      }}
+      onRequestClose={() => {
+        setShowModal(false);
+      }}
+      onSecondarySubmit={() => {
+        setShowModal(false);
+      }}
+      primaryButtonDisabled={disableSubmit}
+    >
+      <ModalBody>
+        <div>
+          <div className="automate_entry_points">
+            <TreeView
+              data={data}
+              aria-label="Automate Entry Points tree"
+              onSelect={(value) => onSelect(value)}
```

Comment:
Minor, but I don't understand why the intermediate function is needed here.  Can this just be 
```suggestion
              onSelect={onSelect}
```

Reply:
Same question for the onExpand